File: raxis/live-e2e/seed/repo/rich-multilang-001/ts-pkg/src/greet.ts
/**
 * Greet a named user. Mirror of `rust-crate::greeting::render_greeting`
 * and `sample_py.greet.render_greeting`.
 *
 * Renaming this function (or its signature) is part of the
 * cross-file-refactor e2e scenario and must propagate to the Rust
 * and Python trees in the same commit.
 */
export function greet(name: string): string {
  const trimmed = name.trim();
  const who = trimmed.length === 0 ? "friend" : trimmed;
  return `Hello, ${who}!`;
}

File: raxis/live-e2e/seed/repo/rich-multilang-001/py-pkg/src/sample_py/greet.py
"""Greeting rendering — mirror of `rust-crate::greeting::render_greeting`.

Renaming this function (or its signature) is part of the
cross-file-refactor e2e scenario and must propagate to the Rust and
TypeScript trees in the same commit.
"""

from __future__ import annotations


def render_greeting(name: str) -> str:
    """Render a greeting for the supplied name.

    An empty name is rendered as the literal "friend" so callers do
    not have to special-case unauthenticated paths.
    """
    trimmed = name.strip()
    who = trimmed if trimmed else "friend"
    return f"Hello, {who}!"
